### PR TITLE
Issue #2991: SummaryJavadoc custom period not always working

### PR DESCRIPTION
**Pull request by sagar-shah94**
Issue: #2991

### CODE REVIEW

**Comment by romani** [comment]
@sagarshah94 , CIs should be green

**Comment by sagar-shah94** [comment]
@romani CI are failing because in some javadoc first line is missing.

**Comment by rnveach** [comment]
@sagar-shah94 You should start by reviewing the violations and see if any are valid. We will require CI to be green to merge.

> Checkstyle failed: Got 328 errors and 0 warnings.

With so many errors, I am thinking most are not correct.

> [checkstyle] [ERROR] /home/travis/build/checkstyle/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/utils/TokenUtils.java:82: First sentence of Javadoc is incomplete (period is missing) or not present. [SummaryJavadoc]

https://github.com/checkstyle/checkstyle/blob/257d13b6bb53245d2491e50d4ca8e39ddfc80035/src/main/java/com/puppycrawl/tools/checkstyle/utils/TokenUtils.java#L82
First one I looked at and this is false. Sentence does end with a period.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java [hunk 1]
```diff
@@ -152,13 +152,34 @@ public void visitJavadocToken(DetailNode ast) {
...
             }
         }
     }
 
     /**
+     * Finds if period is placed at the end of sentence.
+     * @param ast Javadoc root node.
+     * @return true if period is at the end of sentence or false.
+     */
+    private boolean isPeridAtEndOfSentence(DetailNode ast) {
```
Comment (MEZk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java]: @sagar-shah94 
The logic in this method is not correct.
Hint: use JavadocUtils.findFirstToken

**Comment by romani** [comment]
@sagar-shah94 ,  ping.

**Comment by sagar-shah94** [comment]
@romani
https://github.com/checkstyle/checkstyle/pull/4047 this issue will be fixed along with this PR

**Comment by romani** [comment]
@sagar-shah94 , in this case close this PR. and make sure that in #4047 both use cases are covered

**Comment by sagar-shah94** [comment]
@romani ok will fix suggested changes and soon fix both issue with that PR

**Comment by romani** [comment]
@sagar-shah94 , ping.

**Comment by romani** [comment]
@sagar-shah94 , can we finish this PR ?

**Comment by sagar-shah94** [comment]
@romani this issue will be fixed with following PR.
https://github.com/checkstyle/checkstyle/pull/4047

**Comment by romani** [comment]
@sagar-shah94 , please close this PR, if it completely covered by another PR.

**Pull request closed without merging**